src/tools/ceph_dedup/ceph_dedup_tool.cc

```diff
@@ -1040,10 +1040,8 @@ int make_dedup_object(const po::variables_map &opts)
       op.list_snaps(&snap_set, &snap_ret);
       io_ctx.operate(object_name, &op, NULL);
 
-      for (std::vector<librados::clone_info_t>::const_iterator r = snap_set.clones.begin();
-	r != snap_set.clones.end();
-	++r) {
-	io_ctx.snap_set_read(r->cloneid);
+      for (auto& clone : std::as_const(snap_set.clones)) {
```

Comment:
Nit: This is less conventional and also inconsistent with `for (const auto& clone : snap_set.clones) {` style used in the other hunk.

Reply:
sure. will change in the next revision.